File: eRents.UI/e_rents_desktop/lib/models/tenant_preference.dart
import 'package:flutter/material.dart';

class TenantPreference {
  final String id;
  final String userId;
  final DateTime searchStartDate;
  final DateTime? searchEndDate;
  final double? minPrice;
  final double? maxPrice;
  final String city;
  final List<String> amenities;
  final String description;
  final bool isActive;

  TenantPreference({
    required this.id,
    required this.userId,
    required this.searchStartDate,
    this.searchEndDate,
    this.minPrice,
    this.maxPrice,
    required this.city,
    required this.amenities,
    required this.description,
    this.isActive = true,
  });

  factory TenantPreference.fromJson(Map<String, dynamic> json) {
    return TenantPreference(
      id: json['id'],
      userId: json['userId'],
      searchStartDate: DateTime.parse(json['searchStartDate']),
      searchEndDate:
          json['searchEndDate'] != null
              ? DateTime.parse(json['searchEndDate'])
              : null,
      minPrice: json['minPrice'],
      maxPrice: json['maxPrice'],
      city: json['city'],
      amenities: List<String>.from(json['amenities']),
      description: json['description'],
      isActive: json['isActive'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'userId': userId,
      'searchStartDate': searchStartDate.toIso8601String(),
      'searchEndDate': searchEndDate?.toIso8601String(),
      'minPrice': minPrice,
      'maxPrice': maxPrice,
      'city': city,
      'amenities': amenities,
      'description': description,
      'isActive': isActive,
    };
  }
}

File: eRents.Domain/Models/TenantPreference.cs
using System;
using System.Collections.Generic;
using eRents.Domain.Shared;

namespace eRents.Domain.Models;

public partial class TenantPreference : BaseEntity
{
    public int TenantPreferenceId { get; set; }

    public int UserId { get; set; }

    public DateTime SearchStartDate { get; set; }

    public DateTime? SearchEndDate { get; set; }

    public decimal? MinPrice { get; set; }

    public decimal? MaxPrice { get; set; }

    public string City { get; set; } = null!;

    public string? Description { get; set; }

    public bool IsActive { get; set; }

    public virtual User User { get; set; } = null!;

    public virtual ICollection<Amenity> Amenities { get; set; } = new List<Amenity>();
}